### PR TITLE
Refactor extract functions

### PR DESCRIPTION
**Pull request by akram**

### CODE REVIEW

File: pkg/controller/controllerutil/controllerutil.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+func AddToManager(m manager.Manager) error {
+	for _, f := range AddToManagerFuncs {
+		if err := f(m); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Setup Scheme for a resources
```
Comment (jkhelil) [inline pkg/controller/controllerutil/controllerutil.go:48]: golint may not be happy, should be 
 WatchResourceOrStackError setup scheme for a resource , Are you sure it setup a scheme? may

File: pkg/controller/jenkins/jenkins_controller.go [hunk 2]
```diff
@@ -44,18 +39,7 @@ type ControlledResources struct {
...
-	types.NamespacedName
-}
-type StaticResource struct {
-	Object metav1.Object
-	types.NamespacedName
-}
+
 
 // Add creates a new Jenkins Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
```
Comment (jkhelil) [inline pkg/controller/jenkins/jenkins_controller.go:45]: and start it when the Manager is started

File: pkg/controller/jenkinsimage/jenkinsimage_controller.go [hunk 3]
```diff
@@ -2,28 +2,29 @@ package jenkinsimage
...
+const ImageStreamTagKind = "ImageStreamTag"
+const DockerImageKind = "DockerImage"
 
-const registryHostname = "image-registry.openshift-image-registry.svc:5000"
+const DefaultRegistryHostname = "image-registry.openshift-image-registry.svc:5000"
+const DefaultImageNamespace = "openshift"
+const DefaultJenkinsBaseImage = "jenkins" + ":" + "2"
+const DefaultImageStreamTag = "latest"
 
 var log = logf.Log.WithName("controller_jenkinsimage")
```
Comment (jkhelil) [inline pkg/controller/jenkinsimage/jenkinsimage_controller.go:29]: is this should be jenkinsimage_controller in place of controller_jenkinsimage ??

File: pkg/controller/jenkinsimage/jenkinsimage_controller.go [hunk 4]
```diff
@@ -84,11 +75,6 @@ type ReconcileJenkinsImage struct {
 	scheme *runtime.Scheme
 }
 
-// Reconcile reads that state of the cluster for a JenkinsImage object and makes changes based on the state read
-// and what is in the JenkinsImage.Spec
-// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
-// a Pod as an example
-// Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
```
Comment (jkhelil) [inline pkg/controller/jenkinsimage/jenkinsimage_controller.go:78]: s/Request/request

File: pkg/controller/jenkinsimage/new.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	jenkinsv1alpha1 "github.com/redhat-developer/openshift-jenkins-operator/pkg/apis/jenkins/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// newBuildConfg returns a BuildConfig with binatry source strategy using the source image or imagestream specified in the CR
```
Comment (jkhelil) [inline pkg/controller/jenkinsimage/new.go:11]: s/newBuildConfg/newBuildConfig

File: pkg/controller/jenkinsimage/new.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+						Name: imageName,
+					},
+				},
+			},
+		},
+	}
+	return is
+}
+
+// newBuildConfg returns a BuildConfig with binatry source strategy using the source image or imagestream specified in the CR
```
Comment (jkhelil) [inline pkg/controller/jenkinsimage/new.go:35]: s/newBuildConfg/newBuildConfig

File: pkg/controller/jenkinsimage/new.go (hunk 5 above)
Comment (waveywaves) [reply pkg/controller/jenkinsimage/new.go:11]: +1

Comment (waveywaves) [reply pkg/controller/jenkinsimage/new.go:11]: That's my bad actually

File: pkg/controller/jenkinsimage/jenkinsimage_controller.go (hunk 3 above)
Comment (akram) [reply pkg/controller/jenkinsimage/jenkinsimage_controller.go:29]: fixed

File: pkg/controller/jenkins/jenkins_controller.go [hunk 7]
```diff
@@ -12,20 +11,16 @@ import (
...
-	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 var (
-	logController      = logf.Log.WithName("jenkins/controller.go")
+	logController      = logf.Log.WithName("jenkins/jenkins_controller.go")
```
Comment (waveywaves) [inline pkg/controller/jenkins/jenkins_controller.go:23]: Can we return the name of the file somehow ?

File: pkg/controller/jenkinsimage/new.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
+package jenkinsimage
```
Comment (waveywaves) [inline pkg/controller/jenkinsimage/new.go:1]: can we move the contents of new.go to pkg/controller/jenkinsimage/resources/(individual resource).go

File: pkg/common/messages.go [hunk 9]
```diff
@@ -11,16 +11,6 @@ type Messages struct {
 	Errors   []error
```
Comment (waveywaves) [inline pkg/common/messages.go:11]: change name of this pkg to resemble path pkg/messages/messages.go as it is not a good practice to have ambiguous names.

File: pkg/controller/jenkinsimage/new.go (hunk 8 above)
Comment (waveywaves) [reply pkg/controller/jenkinsimage/new.go:1]: Maybe also do this with the other new.go 
This is a common practice and is good for managing code.

File: pkg/controller/jenkins/jenkins_controller.go (hunk 7 above)
Comment (akram) [reply pkg/controller/jenkins/jenkins_controller.go:23]: I don't know how yet.
As soon as we find out how, we can do it in a next PR.

File: pkg/controller/jenkinsimage/new.go (hunk 8 above)
Comment (akram) [reply pkg/controller/jenkinsimage/new.go:1]: as discussed on slack, a next PR for this. Great

**Pull request merged**